File: ProgPython/ActividadesT2/ejercicios2.1/condicionales/ejercicio3.py
def division (num1, num2):
    if num2 == 0:
        raise NameError("ERROR. EL DIVISOR NO PUEDE SER NUNCA 0")
    else:
        resultado = num1 / num2
        return print(f"El resultado es {resultado}")

#PROGRAMA PRINCIPAL

try:
    num1 = int(input("Introduce un número: "))
    num2 = int(input("Introduce otro número: "))

    division(num1,num2)

except Exception as ex:
    print(f"{ex} Debes introducir un divisor distinto de 0")

File: Kotlin/PRIMERA RELACIÓN DE EJERCICIOS AFRICA/Primeros Ejercicios/src/ejercicio3.kt
fun main(){
    print("Introduce Dividendo: ")
    val a = readln().toIntOrNull() ?: 0

    print("Introduce Divisor: ")
    val b = readln().toIntOrNull() ?: 0

    try{
        val resultado: Float = divide(a.toFloat(), b.toFloat())
        println("El resultado es: $resultado")
    }catch (e: Exception){
        println(e.message)
    }
}



fun divide (a: Float, b: Float): Float{

    if (b==0.0f){
        throw IllegalArgumentException("ERROR: Division por cero")
    }else{
        return (a/b)
    }
}